sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/IcebergScanConfig.java

```diff
@@ -204,13 +233,33 @@ public Builder setTableIdentifier(String... names) {
 
     public abstract Builder setBranch(@Nullable String branch);
 
+    public abstract Builder setKeepFields(@Nullable List<String> fields);
+
+    public abstract Builder setDropFields(@Nullable List<String> fields);
+
     public abstract IcebergScanConfig build();
   }
 
   @VisibleForTesting
   abstract Builder toBuilder();
 
   void validate(Table table) {
+    @Nullable List<String> keep = getKeepFields();
+    @Nullable List<String> drop = getDropFields();
+    if (keep != null || drop != null) {
+      checkArgument(
+          keep == null || drop == null, error("only one of 'keep' or 'drop' can be set."));
+      Set<String> fieldsSpecified = Sets.newHashSet(checkNotNull(drop != null ? drop : keep));
```

Comment:
Readability nit: I honestly think just duping the code is easier to read. Since you have already done the `checkArgument`, just

```java
if (keep == null) {
  fieldsSpecified = all the fields
} else {
  fieldsSpecified = keep
}

if (drop != null) {
  fieldsSpecified = fieldsSpecified setminus drop
}
```

Reply:
Thanks, done